File: src/main/java/com/now/nowbot/qq/message/ReplyMessage.java
package com.now.nowbot.qq.message;

import java.util.Map;

public class ReplyMessage extends Message {
    final long id;
    String text;

    public ReplyMessage(long messageId) {
        id = messageId;
    }

    public ReplyMessage(long messageId, String text) {
        id = messageId;
        this.text = text;
    }

    public long getId() {
        return id;
    }

    public String getText() {
        return text;
    }

    @Override
    public JsonMessage toJson() {
        return new JsonMessage("reply", Map.of("id", id));
    }

    @Override
    public String toString() {
        return STR."[at:\{id}]";
    }
}

File: src/main/java/com/now/nowbot/qq/message/ReplyMessage.kt
package com.now.nowbot.qq.message

class ReplyMessage : Message {
    val id: Long
    val text: String?

    constructor(messageID: Long) {
        id = messageID
        this.text = null
    }

    constructor(messageID: Long, text: String?) {
        id = messageID
        this.text = text
    }

    override fun toJson(): JsonMessage {
        return JsonMessage("reply", mapOf("id" to id))
    }

    override fun toString(): String {
        return "[reply:${id}]"
    }
}
